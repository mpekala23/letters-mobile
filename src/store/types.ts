import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
