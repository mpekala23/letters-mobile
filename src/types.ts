// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
