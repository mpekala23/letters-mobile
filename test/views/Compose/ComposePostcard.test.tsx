import React from "react";
import { ComposePostcardScreen } from "@views";
import { render, fireEvent, toJSON } from "@testing-library/react-native";
import { LetterTypes, LetterStatus } from "types";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const setup = () => {
  const navigation = { navigate: jest.fn() };

  const store = mockStore({
    letter: {
      adding: {
        type: LetterTypes.Letters,
        status: LetterStatus.Draft,
        isDraft: true,
        recipientId: -1,
        message: "",
        photoPath: "",
        letterId: -1,
      },
    },
  });

  const StoreProvider = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    navigation,
    store,
    ...render(<ComposePostcardScreen navigation={navigation} />, {
      wrapper: StoreProvider,
    }),
  };
};

describe("ComposePostcard screen", () => {
  it("should match snapshot", () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });
});
