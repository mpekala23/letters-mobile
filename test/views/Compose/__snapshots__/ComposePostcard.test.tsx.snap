// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposePostcard screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    accessible={false}
    activeOpacity={1}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "column",
              "padding": 16,
            },
            Object {
              "flex": 1,
              "paddingBottom": 50,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Poppins-Medium",
                  },
                  Object {
                    "flex": 1,
                    "fontSize": 14,
                    "marginTop": 8,
                  },
                ]
              }
            >
              To
              : 
              First
            </Text>
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FF7171",
                    "borderRadius": 16,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                    "paddingHorizontal": 14,
                  },
                  Object {},
                  Object {
                    "width": 130,
                  },
                  Object {},
                  Object {
                    "elevation": 4,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.22,
                  },
                ]
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Medium",
                    },
                    Object {
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "700",
                      "padding": 6,
                    },
                    Object {},
                    Object {
                      "fontSize": 14,
                    },
                    Object {},
                  ]
                }
              >
                Need Ideas?
              </Text>
            </TouchableOpacity>
          </View>
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 8,
              "flex": 1,
              "marginBottom": 10,
              "overflow": "hidden",
              "width": "100%",
            }
          }
          testID="parent"
        >
          <View
            testID="valid"
          />
          <View
            testID="unfocused"
          />
          <TextInput
            allowFontScaling={true}
            multiline={true}
            numberOfLines={100}
            placeholder="Tell your loved one about something you did / saw today that reminded you of them!"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "rgba(0,0,0,0)",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#9A9A9A",
                  "flex": 1,
                  "fontSize": 16,
                  "padding": 10,
                  "textAlignVertical": "top",
                  "width": "100%",
                },
                Object {
                  "color": "#282828",
                },
                Object {},
                Object {},
                Object {
                  "flex": 1,
                  "fontSize": 18,
                },
                Object {
                  "fontFamily": "Poppins-Regular",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View />
          <View />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
          <View
            style={
              Object {
                "height": 200,
                "opacity": 1,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Array [
                  Object {
                    "height": 200,
                    "marginHorizontal": 8,
                    "width": 180,
                  },
                  Object {
                    "alignItems": "center",
                    "borderRadius": 4,
                    "justifyContent": "center",
                    "overflow": "hidden",
                  },
                ]
              }
              testID="clickable"
            >
              <View
                accessibilityLabel="media placeholder"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={185}
                  bbWidth={164}
                  fill="none"
                  focusable={false}
                  height={185}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 185,
                        "width": 164,
                      },
                    ]
                  }
                  vbHeight={185}
                  vbWidth={164}
                  width={164}
                  xml="<svg width=\\"164\\" height=\\"185\\" viewBox=\\"0 0 164 185\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<rect x=\\"1.5\\" y=\\"1.5\\" width=\\"161\\" height=\\"182\\" rx=\\"6.5\\" stroke=\\"#CCCCCC\\" stroke-width=\\"3\\"/>
<path d=\\"M81.5 75V110M99 92.5H64\\" stroke=\\"#CCCCCC\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/>
</svg>
"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGRect
                      height={182}
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      rx={6.5}
                      stroke={4291611852}
                      strokeWidth={3}
                      width={161}
                      x={1.5}
                      y={1.5}
                    />
                    <RNSVGPath
                      d="M81.5 75V110M99 92.5H64"
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={4291611852}
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth={3}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "opacity": 0,
              "position": "absolute",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={1}
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#9A9A9A",
                "borderTopWidth": 0.3,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "maxWidth": 120,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#6D6D6D",
                    },
                  ]
                }
              >
                300
                 left
              </Text>
            </View>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "maxWidth": 120,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <Text>
                pic
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "maxWidth": 120,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#6D6D6D",
                    },
                  ]
                }
              >
                done
              </Text>
            </TouchableOpacity>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
